src/Event/AbstractEvent.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Sonata\AdminBundle\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy;
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
+
+if (class_exists(LegacyEventDispatcherProxy::class)) {
+    abstract class AbstractEvent extends ContractsEvent
```

Comment:
this fixes those deprecations:

```
  1x: The "Sonata\AdminBundle\Event\PersistenceEvent" class extends "Symfony\Component\EventDispatcher\Event" that is deprecated since Symfony 4.3, use "Symfony\Contracts\EventDispatcher\Event" instead.

Reply:
Oh right… it looks a lot like a BC-break :thinking:

Reply:
You mean if someone relies on the fact that `PersistenceEvent` (for example) is an instance of `Symfony\Component\EventDispatcher\Event`?

Seems weird if someone is doing that :smile: But in theory it could break such cases... yes.

I can also revert this part of the PR and only keep the (definitely BC free signature update)?

Reply:
I think it's alright… and I think we could mark `AbstractEvent` as `@internal`

Reply:
marked as internal :+1: